luci/luci-app-falter-owm/files/owm.sh

```diff
@@ -0,0 +1,318 @@
+#!/bin/sh
+
+. /lib/functions.sh
+. /usr/share/libubox/jshn.sh
+
+OWM_API_VER="1.0"
+
+printhelp() {
+	printf "owm.sh - Tool for registering routers at openwifimap.net\n
+Options:
+\t--help|-h:\tprint this text
+
+\t--dry-run:\tcheck if owm.lua is working (does not paste any data).
+\t\t\tWith this option you can check for errors in your
+\t\t\tconfiguration and test the transmission of data to
+\t\t\tthe map.\n\n
+If invoked without any options, this tool will try to register
+your node at the community-map and print the servers response.
+To work correctly, this tool will need at least the geo-location
+of the node (check correct execution with --dry-run).
+
+To override the server used by this script, set freifunk.community.owm_api.
+"
+}
+
+# save positional argument, as it would get overwritten otherwise.
+CMD_1="$1"
+if [ -n "$CMD_1" ] && [ "$CMD_1" != "--dry-run" ]; then
+	[ "$CMD_1" != "-h" ] && [ "$CMD_1" != "--help" ] && printf "Unrecognized argument %s.\n\n" "$CMD_1"
+	printhelp
+	exit 1
+fi
+
+
+######################
+#                    #
+#  Collect OWM-Data  #
+#                    #
+######################
+
+# Draft for OLSRv2-Links currently not used
+olsr2_links() {
+	json_select $2
+	json_get_var localIP link_bindto
+	json_get_var remoteIP neighbor_originator
+	remotehost="$(nslookup $remoteIP | grep name | sed -e 's/.*name = \(.*\)/\1/' -e 's/\..*//')"".olsr"
+	# Maybe add some stuff here.
+	json_get_var linkQuality domain_metric_out_raw
+	#json_get_var linkQuality domain_metric_in_raw
+	json_get_var ifName "if"
+	json_select ..
+	olsr2links="$olsr2links$localIP $remoteIP $remotehost $linkQuality $ifName;"
+}
+
+olsr4_links() {
+	json_select $2
+	json_get_var localIP localIP
+	json_get_var remoteIP remoteIP
+	remotehost="$(nslookup $remoteIP | grep name | sed -e 's/.*name = \(.*\)/\1/')"
+	json_get_var linkQuality linkQuality
+	json_get_var olsrInterface olsrInterface
+	json_get_var ifName ifName
+	json_select ..
+	olsr4links="$olsr4links$localIP $remoteIP $remotehost $linkQuality $ifName;"
+}
+
+olsr6_links() {
+	json_select $2
+	json_get_var localIP localIP
+	json_get_var remoteIP remoteIP
+	remotehost="$(nslookup $remoteIP | grep name | sed -e 's/.*name = \(.*\)/\1/')"
+	json_get_var linkQuality linkQuality
+	json_get_var olsrInterface olsrInterface
+	json_get_var ifName ifName
+	json_select ..
+	olsr6links="$olsr6links$localIP $remoteIP $remotehost $linkQuality $ifName;"
+}
+
+# This section is relevant for hopglass statistics feature (isUplink/isHotspot)
+OLSRCONFIG=$(printf "/config" | nc localhost 9090)
```

Comment:
At other places, echo is used instead of printf when piping to nc.  It would be good if only one or the other is used.

Reply:
I changed that to only use printf when piping to nc.